### ISSUE TITLE
[ddc] runtime doesn't support interceptors for native types on some corner cases

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/sdk#53864]
DDC's mechanism to add a Dart API to a JavaScript native class relies on adding members to the native class's prototype and to the native class's constructor. The prototype is used for interface dispatch, while the properties added to the constructor are used for implementing dcalls.

Some browser APIs no longer return instances of a class, but objects created via `Object.create`. For example, this came up a few years ago when changes to Gamepad caused [issues](https://github.com/dart-lang/sdk/issues/43750) when bootstraping apps. Our solution to avoid issues was to bailout if an instance doesn't have it's own prototype and constructor (see [this change](https://dart-review.googlesource.com/c/sdk/+/167220)).

Turns out, there are a family of APIs that have a prototype but no constructor. For example, the FileEntry APIs. But, because we bailout when either is missing, the API is not functional. This is the reason behind https://github.com/dart-lang/sdk/issues/47786, but also the reason why in a recent triage we found most fileapi tests failing in DDC (except in firefox).

We could potentially change the bailout to add the interface dispatch if we have a prototype even though we don't have a constructor. That would help use APIs with interface dispatch, but would quickly fail if you use dcalls. This behavior cliff would be very surprising too. In fact, the fileapi tests that led me to this issue would hit this problem, they use dcalls too.

Given our current goals to move towards interop-based DOM access in `package:web`, I don't believe this is worth changing in DDC at the moment.

In summary, I'm filing this issue for tracking and documentation purposes, but my intend is to close it as "wont-fix" unless we have additional reasons to change directions.
